### PR TITLE
[CRIMAP-573] Upgrade gems August 25th 2023

### PR DESCRIPTION
**Pull request by mseedat-moj**
## Description of change
Upgrades ruby gems and fixes any issues arising as a result

## Link to relevant ticket
https://dsdmoj.atlassian.net/browse/CRIMAP-573

## Notes for reviewer
Rubocop rules now enforce deep indexing / iterations hence spec updates

### CODE REVIEW

File: spec/models/reporting/processed_report_spec.rb [hunk 1]
```diff
@@ -15,11 +15,11 @@
     end
 
     it 'includes the row headers' do
-      expect(rows.map(&:first).map(&:content)).to eq(['Today', 'Yesterday', 'Day before yesterday'])
+      expect(rows.map { |x| x.first.content }).to eq(['Today', 'Yesterday', 'Day before yesterday'])
```
Comment (arthurashman) [inline spec/models/reporting/processed_report_spec.rb:18]: Such a silly tiny thing, but would you be up for changing x to r (for row)?

**Review by arthurashman — APPROVED** [review]
Just the one small thought

File: spec/models/reporting/processed_report_spec.rb (hunk 1 above)
Comment (willmcb) [reply spec/models/reporting/processed_report_spec.rb:18]: Or just "row" 🤔

**Review by willmcb — APPROVED** [review]
👍

**Pull request merged**